### PR TITLE
fix: don't lowercase nesting selector

### LINKED ISSUE #4044: [SCSS] Selector starting with `&` is lowercased
**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA6AZhCACYAdKbbAMgFoBGAIQEMAnXAogXwKZABoQIAHGAS2gBnZKDq0IAdwAKdBMJTUAbhD4ATDiGqCYydNQA2guJwBGtamADWcGAGVuFvlADmyGLQCuxkKohhdBkacTka0MFLmzgC21AGG3gBWggAelOZWNrbUUXAAMk5wcUFcHjDcpQBMRd4OtKHIIIJggsKc3LROMADqajAAFsgAHAAMbeJGXebcDe1woYqFnLRwAI4efMsR1NGxSHrxnEZRfG6e3oJOzvpwAIoeEPDVnDDUJj2q-cgVz+Z8+pcAYQgURiDSg0EWIA8RgAKq95PsjEwmEA)

**Input:**
```scss
.foo {
  &-1Bar {
  }
}
```

**Output:**
```scss
.foo {
  &-1bar {
  }
}
```

**Expected behavior:**

```scss
.foo {
  &-1Bar {
  }
}
```

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #4044

### CODE REVIEW

File: tests/css_case/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -725,4 +738,13 @@ a:after {
 table {
 }
 
+.foo {
+  &-KeepSelector {
+  }
+  &-KeepSelector & .KeepClassSelector {
+  }
+  &-100.200 {
```
Comment (j-f1) [inline tests/css_case/__snapshots__/jsfmt.spec.js.snap]: Is this selector valid?

File: src/language-css/printer-postcss.js [hunk 2]
```diff
@@ -289,15 +289,22 @@ function genericPrint(path, options, print) {
       return adjustStrings(node.value, options);
     }
     case "selector-tag": {
+      const parentNode = path.getParentNode();
+      const index = parentNode && parentNode.nodes.indexOf(node);
```
Comment (j-f1) [inline src/language-css/printer-postcss.js]: I think you can get this with `path.getKey()`.

File: tests/css_case/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (evilebottnawi) [reply tests/css_case/__snapshots__/jsfmt.spec.js.snap]: @j-f1 ,i mead 100\.200 it is valid selector

File: src/language-css/printer-postcss.js (hunk 2 above)
Comment (evilebottnawi) [reply src/language-css/printer-postcss.js]: @j-f1 I can't find `path.getKey()` in `fast-path.js` :confused:

Comment (j-f1) [reply src/language-css/printer-postcss.js]: Oops, it’s `getName`. However, it always returns a string.

Comment (evilebottnawi) [reply src/language-css/printer-postcss.js]: @j-f1 we can't get name preview node without getting index current node, approve this PR?

Comment (j-f1) [reply src/language-css/printer-postcss.js]: @evilebottnawi I didn’t quite understand your comment; can you clarify please?

**Pull request merged**